TextToSpeech/impl/TTSDownloader.cpp

```diff
@@ -57,77 +57,71 @@ void TTSDownloader::download(TTSConfiguration config)
 
 void TTSDownloader::downloadThread()
 {
-   while(active)
-   {	   
-       std::unique_lock<std::mutex> lock(m_queueMutex);
-       m_condition.wait(lock,[this]{return needDownload.load();});
+    while(active)
+    {
+        std::unique_lock<std::mutex> lock(m_queueMutex);
+        m_condition.wait(lock,[this]{return needDownload.load();});
 
-       if(active == false)
-       {
-           //invoked from destructor..no download
-           break;
-       }
-       needDownload = false;
-       std::string tts_request;
-       TTSURLConstructer url;
-       m_objectMutex.lock();
-       tts_request = url.constructURL(m_config,"",true);
-       m_objectMutex.unlock();
-       while(tts_request.empty())
-       {
-          sleep(60);
-          //looks like no internet..wait for 1 min and retry
-          m_objectMutex.lock();
-          tts_request = url.constructURL(m_config,"",true);
-          m_objectMutex.unlock();
-       }
-       if((tts_request.compare("null")) == 0)
-       {
-          //got response ..invalid x-api-key set by app
-          TTSLOG_INFO("TTSDownloader::download got invalid response from server\n");
-          continue;
-       }
-       TTSLOG_INFO("TTSDownloader::download going to download file from location %s\n",tts_request.c_str());
-       downloadFile(tts_request);
+        if(active == false)
+        {
+            //invoked from destructor..no download
+            break;
+        }
+        needDownload = false;
+        std::string tts_request;
+        TTSURLConstructer url;
```

Comment:
TTSURLConstructor is been used from multiple areas(downloading/playback)
https://github.com/rdkcentral/rdkservices/blob/sprint/23Q3/TextToSpeech/impl/TTSDownloader.cpp#L72
https://github.com/rdkcentral/rdkservices/blob/sprint/23Q3/TextToSpeech/impl/TTSSpeaker.cpp#L890
also request can happen simultaneously
I think better to leave the implementation as it

Reply:
Yeah, this can be reviewed later if we get chance to see, if the methods can be implemented in a way we don't need to create an instance of TTSURLConstructor everytime.